### ISSUE TITLE
nimble publish SSL issue.

### ISSUE THREAD

**Issue by treeform** [nim-lang/nimble#461]
```
nimble publish
       Info Hint: used config file '/p/Nim/config/nim.cfg' [Conf]
       Info p/Nim/lib/system.nim(470, 35) Warning: unknown magic 'Exception' might crash the compiler [UnknownMagic]
       Info p/Nim/lib/system/nimscript.nim(15, 19) Warning: unknown magic 'BuildOS' might crash the compiler [UnknownMagic]
      Info: Please create a new personal access token on Github in order to allow Nimble to fork the packages repository.
      Hint: Make sure to give the access token access to public repos (public_repo scope)!
      Info: Your default browser should open with the following URL: https://github.com/settings/tokens/new
    Prompt: Personal access token?
    Answer: *******************************
      Info: Writing access token to file:/Users/me/.nimble/github_api_token
Error: unhandled exception: error:1407742E:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert protocol version [SslError]```

Maybe my nimble was not compiled with right SSL flags? Not sure what to do?

```nimble -v 
nimble v0.8.8 compiled at 2017-11-27 18:01:37
git hash: f85219872421b47e6b4a8bcbc7c317c9d1304e2d
```

**Comment by dom96** [comment]
What OS is this?

**Comment by treeform** [comment]
Osx

**Comment by dom96** [comment]
Not a Nimble bug. You need to update your openssl using Homebrew or otherwise.

**Comment by dom96** [comment]
I improved the error message: https://github.com/nim-lang/Nim/commit/1975998c566d2235cad216b2addb2246b8ee95f0

**Comment by treeform** [comment]
For reference, fixed following this comment: https://github.com/dotnet/coreclr/issues/6536#issuecomment-289830546

```
brew update
brew install openssl
mkdir -p /usr/local/lib
ln -s /usr/local/opt/openssl/lib/libcrypto.1.0.0.dylib /usr/local/lib/
ln -s /usr/local/opt/openssl/lib/libssl.1.0.0.dylib /usr/local/lib/
```

**Comment by coocheenin** [comment]
> For reference, fixed following this comment: [dotnet/coreclr#6536 (comment)](https://github.com/dotnet/coreclr/issues/6536#issuecomment-289830546)
> 
> ```
> brew update
> brew install openssl
> mkdir -p /usr/local/lib
> ln -s /usr/local/opt/openssl/lib/libcrypto.1.0.0.dylib /usr/local/lib/
> ln -s /usr/local/opt/openssl/lib/libssl.1.0.0.dylib /usr/local/lib/
> ```

You make my day! The best receipt for a newbie. It would be great, if this code will included to [troubleshooting](https://github.com/nim-lang/nimble#troubleshooting) block. Or I'm wrong, @dom96?

**Comment by dom96** [comment]
We already have something similar in the troubleshooting. You can just set DYLD_LIBRARY_PATH, feel free to add more details in there and link to this issue.
